pkg/api/getambassador.io/v2/handwritten.conversion.go

```diff
@@ -658,3 +659,55 @@ func Convert_v3alpha1_TCPMappingSpec_To_v2_TCPMappingSpec(in *v3alpha1.TCPMappin
 
 	return nil
 }
+
+func Convert_v2_TracingServiceSpec_To_v3alpha1_TracingServiceSpec(in *TracingServiceSpec, out *v3alpha1.TracingServiceSpec, s conversion.Scope) error {
+	if err := autoConvert_v2_TracingServiceSpec_To_v3alpha1_TracingServiceSpec(in, out, s); err != nil {
+		return err
+	}
+	// WARNING: in.TagHeaders requires manual conversion: does not exist in peer-type
+	// if only tag_headers are set, translate to custom_tags.
+	// if both are set, ignore tag_headers.
+	if in.TagHeaders != nil {
```

Comment:
lets reduce the nesting since we only apply this logic when these two things are true:

`if in.TagHeaders != nil && in.V3CustomTags == nil {`